connect/src/main/scala/org/apache/spark/sql/connect/package.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql
```

Comment:
The dsl should not be a public API. Can we move it to `org.apache.spark.sql.catalyst` which is a private package?

Reply:
hmmm that will be a dependency issue? connect depends on sql but this DSL needs to depends on connect proto?

Reply:
Are you actually saying move to `org.apache.spark.sql.catalyst` in connect module?

Reply:
yup

Reply:
BTW, I don't think we have to make it available in spark-shell. The catalyst dsl was added before we have DataFrame API. At that time, it was the only way for end-users to build query plan with a dataframe-like API.

Reply:
@cloud-fan would you prefer that I revert the change for the shell (plus removing those java doc about how to use shell to play with this DSL)?

Reply:
I addressed the package refactoring comment.  Please let me know if you also want to revert the shell change.